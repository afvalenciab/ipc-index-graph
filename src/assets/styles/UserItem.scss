.user--item {
  display: grid;
  grid-template-columns: 3fr 2fr 1fr 1fr;
  grid-gap: 10px;
  box-sizing: border-box;
  align-items: center;
  padding: 15px 0;
  font-weight: 300;

  p {
    margin: 0;
  }

  .user--item_email {
    font-size: 16px;
  }

  button {
    border: 0;
    outline: none;
    color: #fff;
    background: rgba(165, 38, 42,.8);
    padding: 5px 0;
    border-radius: 5px;
    letter-spacing: 1.2px;
    cursor: pointer;
    transition: .3s;
    will-change: transform;
    width: 90%;
  }
  button:hover {
    background: rgba(165, 38, 42, 1);
  }
  button:active {
    transform: scale(.95);
  }
}

@media screen and (max-width: 750px) {
  .user--item {
    .user--item_email {
      font-size: 14px;
    }
  }
}

@media screen and (max-width: 600px) {
  .user--item {
    grid-template-columns: 3fr 1fr 1fr 1fr;
  }
}

@media screen and (max-width: 400px) {
  .user--item {
    .user--item_email {
      font-size: 12px;
    }
  }
}