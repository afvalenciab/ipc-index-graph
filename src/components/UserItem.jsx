import React from 'react';
import { connect } from 'react-redux';
import { updateUser, deleteUser } from '../actions';
import '../assets/styles/UserItem.scss';

const UserItem = (props) => {
  const { _id, email, wrongPass, isAdmin, updateUser, user, deleteUser } = props;

  const handleCheckAdmin = (event) => {
    updateUser(user, _id, { isAdmin: event.target.checked });
  };

  const handleClickDelete = () => {
    deleteUser(user, _id);
  };

  return (
    <div className='user--item'>
      <p className='user--item_email'>{email}</p>
      <p className='user--item_wrong'>{wrongPass}</p>
      <input
        type='checkbox'
        name='admin'
        id='admin'
        className='user--item_input'
        checked={isAdmin}
        onChange={handleCheckAdmin}
      />
      <button type='button' onClick={handleClickDelete}>Eliminar</button>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = {
  updateUser,
  deleteUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(UserItem);
