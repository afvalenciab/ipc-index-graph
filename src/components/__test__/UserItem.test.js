import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import UserItem from '../UserItem';

describe('UserItem testing', () => {
  test('Match Snapshot', () => {
    const userItem = create(
      <ProviderMock>
        <UserItem />
      </ProviderMock>
    );
    expect(userItem.toJSON()).toMatchSnapshot();
  });
});
