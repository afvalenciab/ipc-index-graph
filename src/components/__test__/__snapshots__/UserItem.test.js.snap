// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem testing Match Snapshot 1`] = `
<div
  className="user--item"
>
  <p
    className="user--item_email"
  />
  <p
    className="user--item_wrong"
  />
  <input
    className="user--item_input"
    id="admin"
    name="admin"
    onChange={[Function]}
    type="checkbox"
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Eliminar
  </button>
</div>
`;
